packages/core/parcel/src/cli.js

```diff
@@ -163,6 +164,16 @@ async function run(entries: Array<string>, command: any) {
       }
     });
 
+    command.target = command.target.length > 0 ? command.target : 'browser';
+    if (command.open && command.target === 'browser') {
```

Comment:
`command.target` doesn't do the same thing as in Parcel 1, it build only the list of targets that you specify, not all in `package.json#targets`.

Also, the browser should only open in serve mode, not with `parcel watch`.

The port is already determined in `normalizeOptions` below, `command.port` without any fallbacks is sufficient.

Reply:
`command.port` it's giving `undefined` that why I used: 
`const port = command.port || process.env.PORT || 1234;`

Reply:
The actual port is in the return value of  `normalizeOptions` (in `.serve.port`):

https://github.com/parcel-bundler/parcel/blob/cf3e84cbfc5d3f4a18da9fdcdacaae96c75b5550/packages/core/parcel/src/cli.js#L156

https://github.com/parcel-bundler/parcel/blob/cf3e84cbfc5d3f4a18da9fdcdacaae96c75b5550/packages/core/parcel/src/cli.js#L230-L245

Reply:
Because `options` are private in `Parcel` class, I added `getOptions` so I can access to `port, https, host`, if you have better solution please provide it so we can improve it!